cli/command/container/run.go

```diff
@@ -178,7 +178,10 @@ func runContainer(ctx context.Context, dockerCli command.Cli, runOpts *runOption
 			detachKeys = runOpts.detachKeys
 		}
 
-		closeFn, err := attachContainer(ctx, dockerCli, containerID, &errCh, config, container.AttachOptions{
+		// ctx should not be cancellable here, as this would kill the stream to the container
+		// and we want to keep the stream open until the process in the container exits or until
+		// the user forcefully terminates the CLI.
+		closeFn, err := attachContainer(context.WithoutCancel(ctx), dockerCli, containerID, &errCh, config, container.AttachOptions{
```

Comment:
Looks like the only calls that are still given the cancellable context are:

- createContainer - could cancelling the context here cause an orphan container that's not removed? 🤔 
- ContainerStart - ditto, if cancellation happens before the engine gets the chance to actually attempt the "start" operation, we could block on `errCh` or `statusChan`
- MonitorTtySize - has a loop which doesn't end on context cancellation anyway


I think it would make sense to just strip the cancellation from the original passed context:
```go
func runContainer ...
    ctx := context.WithouCancel(ctx)
    ...
```

Reply:
yeah I think we'll need to see how we should refactor this code in a follow-up. It's a bit of a mess and some of the logic is shared between `run`, `attach` and `start`.

Reply:
After looking at the code again, I'm not sure if stripping the cancellation is the best idea - the code is already a bit of a mess and ignoring the context cancellation completely might not be the right move. 

I think let's try to get this in as is and figure out a way to do a proper refactor of the code.

Reply:
I kinda agree with not making too many changes here at once since the PR is for resolving one specific issue. From what I can see the potential issues mentioned, if present, already exist in the code, so I'd prefer if we tackle them separately once we can reproduce them (manually or with some crafted e2e test). 
WDTY?

Reply:
Yeah I agree

Reply:
true, but I think this fix is "good enough". I'm looking into making this function more "context friendly" in a follow-up. Let's not block the fix on a non-issue!

Reply:
Not blocking this one; but it's only a partial fix for the overall issue with this function. 

After wiring up OTEL, which also sneaked cancellable context here, we've already had multiple fixes in code that originates from this function. All of them were patched by using `context.WithoutCancel`.

Now, we do this for another place.
Instead of patching one issue, the correct fix is to restore the context behavior this function worked with for a long time. The intent behind passing context here was to wire up OTEL:
https://github.com/docker/cli/commit/5400a48aafdd853542fa83b70a11c7ae02f30509#diff-9f722157bd3206a95a2f882edb65882f91b1b07cd5903374d49dad1702eaa8c7R121-R143

This is the change that passed context to this function. At the time the context wasn't cancellable, it just had the OTEL values.
That changed later, when the global context passed via cobra command was made cancellable.

This code wasn't prepared at all to handle context cancellation, so this combination of changes resulting in passing a cancellable context is just a bug and is producing buggy behavior.

We should fix the function first, and then think about refactoring it.